addon/-private/system/model/model.js

```diff
@@ -803,9 +803,9 @@ var Model = Ember.Object.extend(Ember.Evented, {
   willDestroy() {
     //TODO Move!
     this._super(...arguments);
-    this._internalModel.clearRelationships();
-    this._internalModel.recordObjectWillDestroy();
-    //TODO should we set internalModel to null here?
+
+    this._internalModel._resetRecord();
```

Comment:
is there a point to resetting the internal model here, doesn't it just  "go dead" once destroyed?